tests/performance/density.go

```diff
@@ -0,0 +1,123 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021 Red Hat, Inc.
+ *
+ */
+
+package performance
+
+import (
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	k8sv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	kvv1 "kubevirt.io/client-go/api/v1"
+	cd "kubevirt.io/kubevirt/tests/containerdisk"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	"kubevirt.io/client-go/kubecli"
+	"kubevirt.io/kubevirt/tests"
+	"kubevirt.io/kubevirt/tests/util"
+)
+
+var _ = SIGDescribe("Control Plane Performance Density Testing", func() {
+	var (
+		err        error
+		virtClient kubecli.KubevirtClient
+	)
+
+	BeforeEach(func() {
+		if !RunPerfTests {
+			Skip("Performance tests are not enabled.")
+		}
+		virtClient, err = kubecli.GetKubevirtClient()
+		util.PanicOnError(err)
+		tests.BeforeTestCleanup()
+	})
+
+	AfterEach(func() {
+		// ensure the metrics get scraped and collected  till the end
+		time.Sleep(30 * time.Second)
```

Comment:
I don't fully understand this. Where do we collect the metrics so that we need to wait 30s?

Reply:
Humm right, I can improve the description here.
This is related how Prometheus is collecting the data. By default, it scrape metrics with an interval of 30s. That is why we have this "magical" number of 30s. 
So, we sleep 30s, to make sure the all VMI metrics will be scraped by Prometheus.